### ISSUE TITLE
Offline development support

### ISSUE THREAD

**Issue by miljan-aleksic** [gridsome/gridsome#723]
## Summary

When starting the a local develpment server the source data will be fetched and processed, but if there is no internet at the moment the process will fail. While this make sense for the first time, it doens't the second as the data is already there and didn't changed.

### Basic example

This is the time to make breaking changes, so I would suggest making sure the `gridsome develop` always tries to use the cached data first, and if it fails try to rebuild it. Adding an argument `--force-data-fetch` would solve situations where that is desirable.

### Motivation

I work a lot while travelling and most times than desirable I have no internet around, or if I do is slow or limited.

**Comment by hacknug** [comment]
> I would like to suggest to change the current workflow and make sure `gridsome develop` always tries to use the cached data first

I think it would be better to check if there's no Internet, and only then keep the cached data and start the server, no? Then maybe the flag should be to keep the cache instead of removing it.

**Comment by miljan-aleksic** [comment]
Thanks for your inputs @hacknug. Many times I work using my mobile data or find my self with poor speed. Being online and being able to use it effectivly is not always the same.

Perhaps a simple boolean setting in the config file could alter this behaviour.

**Comment by hacknug** [comment]
Yeah, I agree. Maybe I explained myself poorly. I meant that imho the flag should be used to keep the cached data instead of removing it (maybe both could be introduced).

Default should still be what we have now except for when you have no Internet. In that case data coming from sources other than the filesystem should be kept.

Does it make sense?

**Comment by miljan-aleksic** [comment]
Yes, it does :)

**Comment by hjvedvik** [comment]
@miljan-aleksic Offline support would be great :) I don't think it will be easy to automatically cache data for all source plugins because they might have resolver functions etc in the `api.loadSource()` hook that is necessary for the GraphQL schema. But we could force source plugins to only modify the schema in the `api.createSchema()` hook in order to skip the `api.loadSource()` step.

Another approach could be to create a cache API to let source plugins implement caching themself.

And I also think checking the internet connection before loading cached data could be a good idea + an argument for the CLI to force loading cached data.

**Comment by ryanpwaldon** [comment]
Any news on this one?

I'm pulling in data from a large dataset on Airtable. At the moment it's taking about ~2 mins just to start the dev server. Would be super helpful to reuse data from cache!

Anyone know any workarounds?
